newIDE/app/src/Utils/Stash.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * The `Stash` module implements a simple associative store for arbitrary data.
+ */
+
+/**
+ * Constructs a new `Stash` instance, which is essentially a function
+ * with the following signature:
+ *   stash(key: any, value: ?any) -> Stash
+ * Called with a single argument (the `key`), it returns the associated
+ * value, or (re)associates the `key` with given `value` otherwise.
+ *
+ * `stash.peek(key, fallback)`:
+ *   If available, returns the `value` associated with given `key`, or
+ *   the provided `fallback` value otherwise.
+ */
+export default function() {
+  const data = new Map();
+  
+  stash.peek = (key, fallback) => {
+    return data.has(key) ? data.get(key) : fallback;
+  };
+  return stash;
+  
+  function stash(key, value) {
+    if (arguments.length < 2) {
+      return data.get(key);
+    }
```

Comment:
These kind of functions that do different things according to arguments make static typing difficult, and also I would say a bit more difficult to read for the reader. Instead I would favour two separate functions `get`/`set` - or better, could we remove this utility entirely and use a plain old JS object, or a ES6 `Map`?